Makefile

```diff
@@ -1,53 +1,82 @@
 
+.PHONY: all
 all: build
 
-format:
-	@echo "Running gofmt..."
-	@gofmt -w -s -r '(a) -> a' $(SRC_FILES)
-	@echo "Running goimports..."
-	@goimports -e=true -w=true $(SRC_FILES)
-
-generate:
-	@mkdir -p ./mocks
-	@echo "Generating..."
-	@go generate $(PACKAGES)
-
-build: format
-	@echo "Building packages..."
-	@go install $(GO_BUILDFLAGS) $(PACKAGES)
-
-test: build
-	@echo "Running unit tests..."
-	@rm -rf ./coverage.out
-	@mkdir -p ./coverage.out
-	@$(foreach pkg,$(PACKAGES), \
-		go test $(GO_TEST_FLAGS) -cover -coverprofile \
-			./coverage.out/$(shell echo $(pkg) | \
-		sed -e 's|/|_|g' -e 's|^.||g').coverage $(pkg);)
-	@echo "mode: set" > ./coverage.out/combined
-	@cat ./coverage.out/*.coverage | \
-		sed -e '/mode:.*/d' >> ./coverage.out/combined
-	@go tool cover -html ./coverage.out/combined -o ./coverage.html
-	@go tool cover -func=./coverage.out/combined | grep 'total' | \
-		awk '{print "coverage (all): " $$3 " of statements"; }'
+.PHONY: build
+build: fmtcheck go-lint
+	go build -ldflags "$(GOLDFLAGS)" ./...
 
+.PHONY: depscheck
+depscheck:
+	@echo "==> Checking source code dependencies..."
+	@go mod tidy
+	@git diff --exit-code -- go.mod go.sum || \
+		(echo; echo "Found differences in go.mod/go.sum files. Run 'go mod tidy' or revert go.mod/go.sum changes."; exit 1)
+	@# reset go.sum to state before checking if it is clean
+	@git checkout -q go.sum
+
+.PHONY: benchmark
 benchmark:
-	@echo "Running benchmarks..."
-	@$(foreach pkg,$(PACKAGES), \
-		go test -bench="." $(GO_BENCHMARK_FLAGS) $(pkg);)
+	go test -bench=. -benchmem ./...
+
+.PHONY: test
+test: tools
+	go test 				\
+		-race 				\
+		-coverprofile coverage.out 	\
+		./...
+	go tool cover -html=coverage.out -o coverage.html
+
+.PHONY: go-lint
+go-lint: tools
+	@echo "==> Checking source code against linters..."
+	@tools/golangci-lint run ./...
+	@tools/golangci-lint run --build-tags integration ./...
+
+.PHONY: docs-lint
+docs-lint: tools
+	@echo "==> Checking docs against linters..."
+	@tools/misspell -error -source=text README.md || (echo; \
+		echo "Unexpected misspelling found in docs files."; \
+		echo "To automatically fix the misspelling, run 'make docs-lint-fix' and commit the changes."; \
+		exit 1)
+	@docker run -v $(PWD):/markdown 06kellyjac/markdownlint-cli README.md || (echo; \
+		echo "Unexpected issues found in docs Markdown files."; \
+		echo "To apply any automatic fixes, run 'make docs-lint-fix' and commit the changes."; \
+		exit 1)
+
+.PHONY: docs-lint-fix
+docs-lint-fix: tools
+	@echo "==> Applying automatic docs linter fixes..."
+	@tools/misspell -w -source=text docs/
+	@docker run -v $(PWD):/markdown 06kellyjac/markdownlint-cli --fix docs/
+
+.PHONY: lint
+lint: go-lint docs-lint
+
+
+.PHONY: fmt
+fmt:
+	gofmt -s -w .
+
+.PHONY: fmtcheck
+fmtcheck:
+	@./scripts/gofmtcheck.sh
+
+.PHONY: tools
+tools:
+	cd tools && go build -o . github.com/client9/misspell/cmd/misspell
+	cd tools && go build -o . github.com/golangci/golangci-lint/cmd/golangci-lint
```

Comment:
I don't see `.golangci.yml` in this PR, this means we always use the default settings of golangci-lint. Is this intended ?

Reply:
Yes, fully intentional. Default settings are fine for now and out of scope of this PR.